omniledger/service/api.go

```diff
@@ -25,18 +26,29 @@ const ServiceName = "OmniLedger"
 type Client struct {
 	*onet.Client
 	ID      skipchain.SkipBlockID
-	Roster  *onet.Roster
+	Roster  onet.Roster
 	OwnerID darc.Identity
 }
 
-// NewClient instantiates a new Omniledger client.
-// TODO: this needs to be changed to avoid having an invalid Client.
+// NewClientConfig instantiates a new Omniledger client.
+func NewClientConfig(cfg Config) *Client {
+	return &Client{
+		Client:  onet.NewClient(cothority.Suite, ServiceName),
+		ID:      cfg.ID,
+		Roster:  cfg.Roster,
+		OwnerID: cfg.OwnerID,
+	}
+}
+
+// NewClient is like NewClient, but does not close the connection.
 func NewClient() *Client {
-	return &Client{Client: onet.NewClient(cothority.Suite, ServiceName)}
+	log.Warn("Deprecated - should use NewClientConfig")
+	return &Client{Client: onet.NewClientKeep(cothority.Suite, ServiceName)}
 }
 
 // NewClientKeep is like NewClient, but does not close the connection.
 func NewClientKeep() *Client {
+	log.Warn("Deprecated - should use NewClientConfig")
```

Comment:
Nuke it. (But then how will the caller know they should replace omniledger.Client.Client with a client that does connection caching?